File: tests/15_storepc_1.py
#!/usr/bin/env python3

from interp import *

def t():
    c = [
        0xfedcba9876543210,
        0xfffffc0e,
    ]
    i = [
        cpu_encode_op_movib_i64(0,0),
        cpu_encode_op_storepc_i64(0,2),
        cpu_encode_op_movib_i64(1,2),
        cpu_encode_op_break(0),
    ]
    cpu_test("storepc_1", c, i)

t()

File: tests/15_storepc_1.go
package main

import . "glyph/tests/interp"

func t() {
    c := []uint64{
        0xfedcba9876543210,
        0xfffffc0e,
    }
    i := []uint16{
        CPU_encode_op_movq_i64(0,0),
        CPU_encode_op_storepc_i64(0,2),
        CPU_encode_op_movq_i64(1,2),
        CPU_encode_op_break(0),
    }
    CPU_test("storepc_1", c, i);
}

func main() {
    t()
}
